core/mapping/unmarshaler.go

```diff
@@ -403,7 +416,10 @@ func (u *Unmarshaler) processNamedFieldWithValue(field reflect.StructField, valu
 			return fillPrimitive(field.Type, value, mapValue, opts, fullName)
 		}
 
-		return u.processFieldNotFromString(field, value, mapValue, opts, fullName)
+		return u.processFieldNotFromString(field, value, valueWithParent{
+			value:  mapValue,
+			parent: vp.parent,
+		}, opts, fullName)
```

Comment:
Maybe can be used directly the vp variables?

Reply:
fixed.